### PR TITLE
Auto scale workers

### PR DESCRIPTION
**Pull request by pigfall**
* Add script to auto scale worker
* For verifying auto scaling. Added a script to create a env to use remote  build for building tensorflow

### CODE REVIEW

**Review by easeway — COMMENTED** [review]
This is great!

As a customer-facing repository, there are still some work left:

1. Make sure the python script is linted and formatted to community standard;
2. Please DO add enough comments describing what's going on in the script, so customer can understand easily and further modify the script when needed;
3. Allow easier customization: avoid hard code values, try to load from env variables and add validations when needed
4. Can we avoid prom2json (it's easy to parse prom metrics just by line), if required, it should be added in setup.yaml, just to keep the script simplest and minimum.

File: .sandbox/template.yaml [hunk 1]
```diff
@@ -24,6 +24,7 @@ volumes:
   - name: bf-config
     content:
       text: |
+        prometheusPort: 9090
```
Comment (easeway) [inline .sandbox/template.yaml]: This is not necessary as it's already default.

File: .sandbox/auto_scale.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    ))
+    if len(worker_pool_size_metric) > 0:
+        server_metrics.worker_pool_size = functools.reduce(
+            lambda a,b: a+b,
+            worker_pool_size_metric,
+        )
+
+    return server_metrics,None
+
+def scrpae_buildfarm_worker_metrics(worker_name):
```
Comment (easeway) [inline .sandbox/auto_scale.py]: spelling: `scrape`

**Comment by pigfall** [comment]
> This is great!
> 
> As a customer-facing repository, there are still some work left:
> 
> 1. Make sure the python script is linted and formatted to community standard;
> 2. Please DO add enough comments describing what's going on in the script, so customer can understand easily and further modify the script when needed;
> 3. Allow easier customization: avoid hard code values, try to load from env variables and add validations when needed
> 4. Can we avoid prom2json (it's easy to parse prom metrics just by line), if required, it should be added in setup.yaml, just to keep the script simplest and minimum.

Updated. Please review again when you get a chance

File: .sandbox/auto_scale.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    return metrics
+
+
+# Scrape the metrics of a buildfarm worker.
+def scrape_buildfarm_worker_metrics(worker_name):
+    url = "http://{}:9090/metric".format(worker_name)
+    worker_metrics = BuildfarmWorkerMetrics() 
+    worker_metrics.worker_name = worker_name
+
+    result = subprocess.run(["prom2json",url],capture_output=True, text=True)
```
Comment (easeway) [inline .sandbox/auto_scale.py:298]: Where is `prom2json` installed? I saw it was removed from the script but I didn't see it's being added to setup.yaml.

**Pull request merged**